### ISSUE TITLE
How to apply a filter by default?

### ISSUE THREAD

**Issue by dodumosu** [flask-admin/flask-admin#2248]
Hello, I want a list view to have a column filter applied by default, and only to be revoked when the user explicitly resets/removes the filter. Is there a way to get this done?

**Comment by bekab95** [comment]
I think It can be made by passing some parameters to 
`admin.add_view(GenericView(SomeModel, name="Some Model Name", category="Some Category"))`

**Comment by Mo8it** [comment]
Override `get_query` and `get_query_count`. Example:

```python
    def query_modifier(self, query):
        return query # Apply a default filter here

    def get_query(self):
        return self.query_modifier(super().get_query())

    def get_count_query(self):
        return self.query_modifier(super().get_count_query())
```

**Comment by GrenderG** [comment]
But your solution, @Mo8it, would **permanently** enable that filter, no? You wouldn't be able to disable it from the UI.

**Comment by Mo8it** [comment]
@GrenderG, yes, I did just read the issue again and see that the author wants the option to disable it from the UI. Then my solution does not help. Thanks for pointing this out.

**Comment by ElLorans** [comment]
You can override the navbar template to link not only the model view but also the default filter. When the client clicks on the navbar link, they are redirected to the "default" filter, but they can remove it from the GUI.
